### PR TITLE
feat: add modal to pipeline table

### PR DESCRIPTION
**Pull request by rajatgupta24**
- Adds `Name` in the `pipelines` struct
- Adds a confirmation modal for stop pipeline button in pipelines table

### CODE REVIEW

File: web/src/lib/cellRenderer.ts [hunk 1]
```diff
@@ -4,25 +4,35 @@ export const HTMLInColumnCell = (link: string, text: string): string => {
...
+
+    const pipeline = {
       owner,
       branch,
       repository,
       build
-    }));
+    }
+
+    p?.addEventListener('click', () => {
```
Comment (ankitm123) [inline web/src/lib/cellRenderer.ts:29]: Do we need the chaining operator here? Can p be nullish (null or undefined)?

Comment (rajatgupta24) [reply web/src/lib/cellRenderer.ts]: I added that will developing, can be removed.

File: web/src/routes/admin/pipelines/+page.svelte [hunk 2]
```diff
@@ -4,9 +4,58 @@
   import type { GridOptions } from 'ag-grid-community'
   import { BTNInColumnCell, HTMLInColumnCell } from '$lib/cellRenderer'
   import { stopPipelineHandler } from '$lib/Components/Pipelines/StopPipeline'
+  import Modal from '$src/lib/Components/Modal.svelte'
   
   export let data
 
+  let show = false
+  let title = ''
```
[the remark this answers is not in the document]
Comment (rajatgupta24) [reply web/src/routes/admin/pipelines/+page.svelte]: This is kind of the way, it is done in javascript.
Not strict but just the way people do.

Though I think we should add types to them

**Review by ankitm123 — APPROVED** [review]

**Pull request merged**